src/test-utils/index.tsx

```diff
@@ -179,24 +183,64 @@ export const openStoryPage = ({
         isDarkMode,
     });
 
+const hydrateSSRPage = async (page: PageApi): Promise<void> => {
+    await page.evaluate(() => {
+        // @ts-expect-error this executes in the browser, see ssr.tsx
+        window.hydrate();
+    });
+};
+
+const checkHydrationMissmatch = async (page: PageApi): Promise<void> => {
+    const {testPath, currentTestName} = expect.getState();
+    const tmpdir = os.tmpdir();
+    const snapshotId = kebabCase(`${path.basename(testPath)}-${currentTestName}`);
+    const baselineImagePath = path.join(tmpdir, `${snapshotId}-snap.png`);
+    await page.screenshot({path: baselineImagePath});
+
+    await hydrateSSRPage(page);
+    const hydrateImage = await page.screenshot();
+
+    try {
+        expect(hydrateImage).toMatchImageSnapshot({
+            // use the saved ssr snapshot as the baseline
+            customSnapshotsDir: tmpdir,
+            customSnapshotIdentifier: snapshotId,
+            customDiffDir: path.join(
+                __dirname,
+                '..',
+                '__acceptance_tests__',
+                '__image_snapshots__',
+                '__diff_output__'
+            ),
+        });
+    } catch (error: any) {
+        Error.captureStackTrace(error, checkHydrationMissmatch);
+        throw error;
+    } finally {
+        fs.unlinkSync(baselineImagePath);
+    }
+};
```

Comment:
this is how we check there isn't any visual changes between SSRed content and hydrated ui